CRM/Batch/BAO/Batch.php

```diff
@@ -722,24 +723,16 @@ public static function getBatchFinancialItems($entityID, $returnValues, $notPres
     }
     if (!empty($query->_where[0])) {
       $where = implode(' AND ', $query->_where[0]) .
-        " AND civicrm_entity_batch.batch_id IS NULL
-         AND civicrm_entity_financial_trxn.entity_table = 'civicrm_contribution'";
+        " AND civicrm_entity_batch.batch_id IS NULL ";
       $where = str_replace('civicrm_contribution.payment_instrument_id', 'civicrm_financial_trxn.payment_instrument_id', $where);
-      $searchValue = TRUE;
     }
     else {
-      $searchValue = FALSE;
-    }
-
-    if (!$searchValue) {
       if (!$notPresent) {
-        $where = " ( civicrm_entity_batch.batch_id = {$entityID}
-        AND civicrm_entity_batch.entity_table = 'civicrm_financial_trxn'
```

Comment:
This criteria of entity_table = ft seems to replicate https://github.com/civicrm/civicrm-core/pull/9942/files#diff-4edf3d9f312c7543c00ef7d38bf21044L652, which is still present at https://github.com/civicrm/civicrm-core/pull/9942/files#diff-4edf3d9f312c7543c00ef7d38bf21044R654. I haven't followed through logic of $searchValue and $notPresent, but looking at the diff on a computer screen instead of my phone it looks better.